ci/update-packages.sh

```diff
@@ -0,0 +1,53 @@
+#!/bin/bash
+
+######################################################################
+# This script sets up the docker environment for updating packages
+######################################################################
+
+set -e
+set -x
+
+# Check if docker is installed
+if ! [ -x "$(command -v docker)" ]; then
+  echo "Docker is not installed. Please install docker and try again."
+  exit 1
+fi
+
+# Check if we can run docker without sudo
+if ! docker ps > /dev/null 2>&1; then
+  echo "Docker is not running without sudo. Please add your user to the docker group and try again."
+  exit 1
+fi
+
+SOURCE_DIR=$(pwd)/../
+
+TEMP_DIR=$(mktemp -d)
+
+# Copy the current directory to the temp directory
+cp -r ../ $TEMP_DIR
```

Comment:
This is the repo directory (holding the .git folder), not the current directory.

I also think that only a subset of files needs to be copied, such as the files under version management, perhaps using: `git clone .. $TEMP_DIR`?

Reply:
But the issue is we are upgrading dependencies only when we make changes to our flexmeasures repo correct?

Reply:
My issue here is that my repo directory is full of large data files (not under version management) that do not need to be copied. A local git clone was much faster than recursively copying the entire project directory.

But I do see your point that uncommitted changes (e.g. to `app.in`) might not be reflected in the copy. That is your point, right?

Reply:
Yes, that makes it a true developer tool.

In the end, non-git files do not need to be in it I believe.

We had some code back in the day in our to_pypi script which checked if there were non-git files present.

Reply:
Does git have any command to print out the files it is currently tracking in the repo? Maybe copy only those tracked files.

Reply:
https://stackoverflow.com/questions/15606955/how-can-i-make-git-show-a-list-of-the-files-that-are-being-tracked

Alternatively, we could simply warn or raise in case of uncommitted changes. I might even consider that a cleaner approach.

Reply:
I actually believe the requirements folder is all we need.
We are just doing pip-tools commands here. ..

Reply:
We also run tests as part of `make upgrade-deps`, which makes sense to me. That needs the code.

Reply:
But I disabled running tests when upgrading packages using docker.